README.md

```diff
@@ -1,13 +1,184 @@
+    new ServiceProvider("https://horizon.stellar.org", ServiceProvider.NETWORK_ID_MAIN));
+KinClient kinClient = new KinClient(context, infuraProvider);
+```
+
+To connect to a test Stellar network use the following ServiceProvider:
+```java
+new ServiceProvider("https://horizon-testnet.stellar.org", ServiceProvider.NETWORK_ID_TEST)
+``` 
+
+### Creating and retrieving a KIN account
+The first time you use `KinClient` you need to create a new account, using a passphrase. 
+The details of the account created will be securely stored on the device.
+```java
+KinAccount account;
+try {
+    if (!kinClient.hasAccount()) {
+        account = kinClient.addAccount("yourPassphrase");
+    }
+} catch (CreateAccountException e) {
+    e.printStackTrace();
+}
+```
+
+Once an account has been created there is no need to call `createAccount` again on the same device. 
+From then on calling `getAccount` will retrieve the account stored on the device.
+```java
+if (kinClient.hasAccount()) {
+    account = kinClient.getAccount(0);
+}
+``` 
+
+You can delete your account from the device using `deleteAccount` with the passphrase you used to create it as a parameter, 
+but beware! you will lose all your existing KIN if you do this.
+```java
+kinClient.deleteAccount(int index, String passphrase);
+``` 
+
+### Onboarding
+Before an account can be used, it must be created on Stellar blockchain, by a different entity (Server) that has an account 
+on Stellar network.
+and then must the account must be activated, before it can receive or send KIN.
+
+
+```java
+Request<Balance> activationRequest = account.activate(passphrase)
+activationRequest.run(new ResultCallback<Void>() {
+    @Override
+    public void onResult(Void result) {
+        Log.d("example", "Account is activated");
+    }
+
+    @Override
+    public void onError(Exception e) {
+        e.printStackTrace();
+    }
+});
+``` 
+For a complete example of this process, take a look at Sample App `OnBoarding` class.
+
+### Public Address
+Your account can be identified via it's public address. To retrieve the account public address use:
+```java
+account.getPublicAddress();
+```
+
+
+### Retrieving Balance
+To retrieve the balance of your account in KIN call the `getBalance` method: 
+```java
+Request<Balance> balanceRequest = account.getBalance();
+balanceRequest.run(new ResultCallback<Balance>() {
+
+    @Override
+    public void onResult(Balance result) {
+        Log.d("example", "The balance is: " + result.value(2));
+    }
+
+    @Override
+        public void onError(Exception e) {
+            e.printStackTrace();
+        }
+});
+```
+
+### Transfering KIN to another account
+To transfer KIN to another account, you need the public address of the account you want 
+to transfer the KIN to. 
+
+The following code will transfer 20 KIN to account "#AB12349ACF123". 
+```java
+String toAddress = "#AB12349ACF123";
+String passphrase = "yourPassphrase";
+BigDecimal amountInKin = new BigDecimal("20");
```

Comment:
value in Kin? What about fractional values?

Reply:
well, `BigDecimal` has fractional value, so I guess that speaks for itself, as for ranges, it is stellar specific, don't see a reason to mention that (maybe link to stellar doc)